<?xml version="1.0" ?>
<launch>

  <arg name="driver_ns"		default="b_bot/ur_hardware_interface"/>
  <arg name="tool_names"	default="screw_tool_m3
					 screw_tool_m4
					 suction_tool
					 base_fixture"/>
  <arg name="tool_number"	default="0"/>

  <group unless="$(eval tool_number == len(tool_names.split()))">
    <arg name="tool_name"
	 value="$(eval tool_names.split()[tool_number])"/>

    <node name="$(arg tool_name)_controller"
  	  pkg="aist_fastening_tools" type="suction_tool_controller.py"
  	  output="screen">
      <param name="driver_ns"	value="$(arg driver_ns)"/>
      <rosparam file="$(dirname)/../config/$(arg tool_name).yaml"/>
    </node>

    <include file="$(dirname)/suction_tool_controllers.launch">
      <arg name="driver_ns"	value="$(arg driver_ns)"/>
      <arg name="tool_names"	value="$(arg tool_names)"/>
      <arg name="tool_number"	value="$(eval tool_number + 1)"/>
    </include>
  </group>

</launch>
